core/modules/ccontrol/UserQuerySelect.cc

```diff
@@ -379,7 +382,7 @@ void UserQuerySelect::setupChunking() {
 }
 
 // register query in qmeta database
-void UserQuerySelect::qMetaRegister()
+void UserQuerySelect::qMetaRegister(std::string const& resultLocation, std::string const& msgTableName)
```

Comment:
The factory code is a big mess, I'd like to clean it up at some point, but it needs significant refactoring and I do not want to touch it now. The method cannot be called more than once, but can be called zero times I guess this is the main reason why it is outside of constructor (constructor does not have enough info to decide whether to call it or not).